wandb/sdk/internal/internal_api.py

```diff
@@ -1040,13 +1037,77 @@ def create_run_queue(
         ]
         return result
 
+    @normalize_exceptions
+    def push_to_run_queue_by_name(
+        self, entity: str, project: str, queue_name: str, run_spec: str
+    ) -> Optional[Dict[str, Any]]:
+        """
+        Queryless mutation, should be used before legacy fallback method
+        """
+
+        def no_retry_4xx(e: Exception) -> bool:
+            if not isinstance(e, requests.HTTPError):
+                return True
+            if (
+                not (e.response.status_code >= 400 and e.response.status_code < 500)
```

Comment:
nit: python allows you to specify range as follows: `400 <= e.response.status_code < 500`.
Don't have to change jfyi...

Reply:
Yep, thanks! I went ahead and also refactored this function into `utils` as it was now being used in three seperate places.